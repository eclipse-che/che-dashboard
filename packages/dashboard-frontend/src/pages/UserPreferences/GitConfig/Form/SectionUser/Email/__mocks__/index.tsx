/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import * as React from 'react';

import { Props } from '..';

export class GitConfigUserEmail extends React.PureComponent<Props> {
  public render(): React.ReactElement {
    const { onChange, isLoading, value } = this.props;

    return (
      <div data-testid="config-user-email">
        <span data-testid="isLoading">{isLoading}</span>
        <span data-testid="value">{value}</span>
        <button onClick={() => onChange('new-user@che', true)}>Change Email Valid</button>
        <button onClick={() => onChange('new-user@che', false)}>Change Email Invalid</button>
      </div>
    );
  }
}
