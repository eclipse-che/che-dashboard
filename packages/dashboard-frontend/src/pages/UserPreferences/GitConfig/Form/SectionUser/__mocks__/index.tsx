/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import * as React from 'react';

import { Props } from '..';

export class GitConfigSectionUser extends React.PureComponent<Props> {
  render(): React.ReactElement {
    const { config, onChange } = this.props;
    return (
      <div>
        <div data-testid="user-email">{config.user.email}</div>
        <div data-testid="user-name">{config.user.name}</div>
        <button
          onClick={() => onChange({ user: { email: 'new-user@che', name: 'new user' } }, true)}
        >
          Change Email Valid
        </button>
      </div>
    );
  }
}
