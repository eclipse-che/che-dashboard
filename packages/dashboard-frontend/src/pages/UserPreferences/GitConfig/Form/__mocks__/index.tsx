/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '..';

export class GitConfigForm extends React.PureComponent<Props> {
  render(): React.ReactElement {
    const { isLoading, gitConfig, onReload, onSave } = this.props;

    return (
      <div data-testid="git-config-form">
        <span data-testid="is-loading">{isLoading}</span>
        <span data-testid="git-config">{JSON.stringify(gitConfig)}</span>
        <button onClick={() => onSave({ user: { email: 'new-user@che', name: 'new user name' } })}>
          Save Config
        </button>
        <button onClick={() => onReload()}>Reload Config</button>
      </div>
    );
  }
}
