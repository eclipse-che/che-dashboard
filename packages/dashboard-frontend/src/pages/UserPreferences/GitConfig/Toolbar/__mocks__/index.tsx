/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/UserPreferences/GitConfig/Toolbar';

export class GitConfigToolbar extends React.PureComponent<Props> {
  render() {
    const { mode, onAdd, onChangeMode } = this.props;
    const nextMode = mode === 'form' ? 'viewer' : 'form';
    return (
      <div>
        Mock Git Config Toolbar
        <span data-testid="toolbar-mode">{mode}</span>
        <button data-testid="toolbar-on-add" onClick={onAdd}>
          Add
        </button>
        <button data-testid="toolbar-on-change-mode" onClick={() => onChangeMode(nextMode)}>
          Switch Mode
        </button>
      </div>
    );
  }
}
