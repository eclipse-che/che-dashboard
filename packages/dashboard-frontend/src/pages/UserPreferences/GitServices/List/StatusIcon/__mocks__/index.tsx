/*
 * Copyright (c) 2018-2024 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';

import { Props } from '@/pages/UserPreferences/GitServices/List/StatusIcon';

export class GitServiceStatusIcon extends React.PureComponent<Props> {
  public render(): React.ReactElement {
    const { gitProvider } = this.props;

    return (
      <span data-testid="git-service-status-icon">
        <span>Authentication Status Icon for {gitProvider}</span>
      </span>
    );
  }
}
