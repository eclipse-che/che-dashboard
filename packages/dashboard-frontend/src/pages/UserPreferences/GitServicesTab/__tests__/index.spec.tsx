/*
 * Copyright (c) 2018-2023 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import userEvent from '@testing-library/user-event';
import { Provider } from 'react-redux';
import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Store } from 'redux';
import { GitServicesTab } from '..';
import { FakeGitOauthBuilder } from './__mocks__/gitOauthRowBuilder';
import { FakeStoreBuilder } from '../../../../store/__mocks__/storeBuilder';
import { selectIsLoading, selectGitOauth } from '../../../../store/GitOauthConfig/selectors';
import { actionCreators } from '../../../../store/GitOauthConfig';

describe('GitServices', () => {
  const mockRevokeOauth = jest.fn();
  const requestGitOauthConfig = jest.fn();

  const getComponent = (store: Store): React.ReactElement => {
    const state = store.getState();
    const gitOauth = selectGitOauth(state);
    const isLoading = selectIsLoading(state);
    return (
      <Provider store={store}>
        <GitServicesTab
          gitOauth={gitOauth}
          isLoading={isLoading}
          revokeOauth={mockRevokeOauth}
          requestGitOauthConfig={requestGitOauthConfig}
        />
      </Provider>
    );
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should correctly render the component without sit services', () => {
    const component = getComponent(new FakeStoreBuilder().build());
    render(component);

    const emptyStateText = screen.queryByText('No Git Services');
    expect(emptyStateText).toBeTruthy();

    const json = renderer.create(component).toJSON();

    expect(json).toMatchSnapshot();
  });

  it('should correctly render the component which contains two git services', () => {
    const component = getComponent(
      new FakeStoreBuilder()
        .withGitOauthConfig([
          new FakeGitOauthBuilder()
            .withName('github')
            .withEndpointUrl('https://github.com')
            .build(),
          new FakeGitOauthBuilder()
            .withName('gitlab')
            .withEndpointUrl('https://gitlab.com')
            .build(),
        ])
        .build(),
    );
    render(component);

    const emptyStateText = screen.queryByText('No Git Services');
    expect(emptyStateText).not.toBeTruthy();

    const json = renderer.create(component).toJSON();

    expect(json).toMatchSnapshot();
  });

  it('should revoke a git service', () => {
    const spyRevokeOauth = jest.spyOn(actionCreators, 'revokeOauth');
    const component = getComponent(
      new FakeStoreBuilder()
        .withGitOauthConfig([
          new FakeGitOauthBuilder()
            .withName('github')
            .withEndpointUrl('https://github.com')
            .build(),
        ])
        .build(),
    );
    render(component);

    const menuButton = screen.getByLabelText('Actions');
    userEvent.click(menuButton);

    const revokeItem = screen.getByRole('menuitem', { name: /Revoke/i });
    userEvent.click(revokeItem);

    const text = screen.findByText("Would you like to revoke git service 'GitHub'?");
    expect(text).toBeTruthy();

    const revokeButton = screen.getByTestId('revoke-button');
    expect(revokeButton).toBeDisabled();

    const checkbox = screen.getByTestId('warning-info-checkbox');
    userEvent.click(checkbox);
    expect(revokeButton).toBeEnabled();

    userEvent.click(revokeButton);
    expect(spyRevokeOauth).toBeCalledWith('github');
  });
});
