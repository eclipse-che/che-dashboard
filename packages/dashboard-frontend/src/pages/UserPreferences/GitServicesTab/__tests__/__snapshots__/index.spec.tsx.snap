// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitServices should correctly render the component which contains four git services 1`] = `
Array [
  <span
    className="progressLine"
  >
    
  </span>,
  <section
    className="pf-c-page__main-section"
  >
    <div
      className="pf-c-toolbar pf-m-page-insets"
      id="git-services-list-toolbar"
    >
      <div
        className="pf-c-toolbar__content"
      >
        <div
          className="pf-c-toolbar__content-section"
        >
          <div
            className="pf-c-toolbar__item"
          >
            <button
              aria-disabled={true}
              aria-label={null}
              className="pf-c-button pf-m-danger pf-m-disabled"
              data-ouia-component-id="OUIA-Generated-Button-danger-2"
              data-ouia-component-type="PF4/Button"
              data-ouia-safe={true}
              data-testid="bulk-revoke-button"
              disabled={true}
              onClick={[Function]}
              role={null}
              tabIndex={null}
              type="button"
            >
              Revoke
            </button>
          </div>
        </div>
        <div
          className="pf-c-toolbar__expandable-content"
          id="git-services-list-toolbar-expandable-content-1"
        >
          <div
            className="pf-c-toolbar__group"
          />
        </div>
      </div>
      <div
        className="pf-c-toolbar__content pf-m-hidden"
        hidden={true}
      >
        <div
          className="pf-c-toolbar__group"
        />
      </div>
    </div>
    <table
      aria-label="Git services"
      className="pf-c-table pf-m-grid-md pf-m-compact"
      data-ouia-component-id="OUIA-Generated-Table-4"
      data-ouia-component-type="PF4/Table"
      data-ouia-safe={true}
      role="grid"
    >
      <thead
        className=""
      >
        <tr
          className=""
          data-ouia-component-id="OUIA-Generated-TableRow-6"
          data-ouia-component-type="PF4/TableRow"
          data-ouia-safe={true}
          hidden={false}
        >
          <th
            className=""
            data-key={0}
            data-label=""
            onMouseEnter={[Function]}
            scope={null}
          >
            
          </th>
          <th
            className=""
            data-key={1}
            data-label="Name"
            onMouseEnter={[Function]}
            scope="col"
          >
            Name
          </th>
          <th
            className=""
            data-key={2}
            data-label="Server"
            onMouseEnter={[Function]}
            scope="col"
          >
            Server
          </th>
          <td
            className=""
            data-key={3}
            data-label=""
            onMouseEnter={[Function]}
            scope={null}
          >
            
          </td>
        </tr>
      </thead>
      <tbody
        className=""
        role="rowgroup"
      >
        <tr
          className=""
          data-ouia-component-id="OUIA-Generated-TableRow-7"
          data-ouia-component-type="PF4/TableRow"
          data-ouia-safe={true}
          hidden={false}
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <td
            className="pf-c-table__check"
            data-key={0}
            data-label={null}
            onMouseEnter={[Function]}
          >
            <label>
              <input
                aria-label="Select row 0"
                checked={false}
                name="checkrow0"
                onChange={[Function]}
                type="checkbox"
              />
            </label>
          </td>
          <td
            className=""
            data-key={1}
            data-label="Name"
            onMouseEnter={[Function]}
          >
            GitHub
          </td>
          <td
            className=""
            data-key={2}
            data-label="Server"
            onMouseEnter={[Function]}
          >
            <a
              href="https://github.dummy.endpoint.com"
              rel="noreferrer"
              target="_blank"
            >
              https://github.dummy.endpoint.com
            </a>
          </td>
          <td
            className="pf-c-table__action"
            data-key={3}
            data-label={null}
            onMouseEnter={[Function]}
            style={
              Object {
                "paddingRight": 0,
              }
            }
          >
            <div
              className="pf-c-dropdown pf-m-align-right"
              data-ouia-component-id="OUIA-Generated-Dropdown-5"
              data-ouia-component-type="PF4/Dropdown"
              data-ouia-safe={true}
            >
              <button
                aria-expanded={false}
                aria-haspopup={true}
                aria-label="Actions"
                className="pf-c-dropdown__toggle pf-m-plain"
                disabled={false}
                id="pf-dropdown-toggle-id-4"
                onClick={[Function]}
                onKeyDown={[Function]}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style={
                    Object {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 192 512"
                  width="1em"
                >
                  <path
                    d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                  />
                </svg>
              </button>
            </div>
          </td>
        </tr>
        <tr
          className=""
          data-ouia-component-id="OUIA-Generated-TableRow-8"
          data-ouia-component-type="PF4/TableRow"
          data-ouia-safe={true}
          hidden={false}
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <td
            className="pf-c-table__check"
            data-key={0}
            data-label={null}
            onMouseEnter={[Function]}
          >
            <label>
              <input
                aria-label="Select row 1"
                checked={false}
                disabled={true}
                name="checkrow1"
                onChange={[Function]}
                type="checkbox"
              />
            </label>
          </td>
          <td
            className=""
            data-key={1}
            data-label="Name"
            onMouseEnter={[Function]}
          >
            GitLab
            <span
              data-testid="gitlab-unsupported-automatic-revoke-tooltip"
              data-tip="Provided API does not support the automatic token revoke. You can revoke it manually."
            >
              <svg
                aria-hidden={true}
                aria-labelledby={null}
                fill="var(--pf-global--warning-color--100)"
                height="1em"
                role="img"
                style={
                  Object {
                    "margin": "2px 5px",
                    "verticalAlign": "text-top",
                  }
                }
                viewBox="0 0 1088 1024"
                width="1em"
              >
                <path
                  d="M1057.10141,663.5 L845.101405,215.4 C787.101405,71.8 665.401405,0 542.901405,0 C420.201405,0 296.701405,71.9 235.001405,215.6 L31.7014051,648.5 C10.4014051,700 -0.0985948775,752.3 0.000697596367,800.8 C0.301405123,924.8 70.2014051,1024 209.101405,1024 L868.401405,1024 C1005.80141,1024 1087.70141,918.6 1088.00215,795.5 C1088.10141,752.4 1078.20141,707.2 1057.10141,663.5 Z M959.401405,800.3 C958.701405,822.9 952.901405,843.5 942.601405,859.7 C926.801405,884.6 902.601405,896.7 868.301405,896.7 L209.101405,896.7 C191.201405,896.7 176.601405,893.8 165.401405,888.2 C157.301405,884 150.801405,878.4 145.401405,870.3 C135.101405,855 129.101405,832 128.401405,805.6 C127.601405,772.8 134.901405,736.5 149.401405,700.5 L353.001405,266.7 C363.201405,242.9 376.101405,221.5 391.101405,203.2 C404.801405,186.6 420.301405,172.4 437.401405,161.1 C469.201405,139.9 505.701405,128.8 542.901405,128.8 C579.701405,128.8 615.401405,139.8 646.001405,160.5 C662.401405,171.6 677.101405,185.4 690.101405,201.6 C704.501405,219.6 716.401405,240.6 725.901405,264 L940.901405,718.9 L941.101405,719.3 L941.301405,719.7 C953.901405,746 960.201405,773.9 959.401405,800.3 Z M586.601405,832 L501.301405,832 C489.501405,831.8 480.201405,821.5 480.001405,808.7 L480.001405,727.3 C480.201405,714.5 489.601405,704.3 501.301405,704 L586.601405,704 C598.401405,704.2 607.701405,714.5 607.901405,727.3 L607.901405,808.7 L608.001405,808.7 C607.701405,821.5 598.301405,831.8 586.601405,832 M639.901405,290.7 L613.201405,610.4 C611.801405,626.9 598.001405,640 581.301405,640 L506.601405,640 C490.001405,640 476.101405,627.2 474.701405,610.7 L448.101405,291 C446.501405,272.3 461.301405,256.3 480.001405,256.3 L608.001405,256 C626.701405,256 641.401405,272 639.901405,290.7"
                />
              </svg>
            </span>
            <div>
              Dummy Tooltip
            </div>
          </td>
          <td
            className=""
            data-key={2}
            data-label="Server"
            onMouseEnter={[Function]}
          >
            <a
              href="https://gitlab.dummy.endpoint.com"
              rel="noreferrer"
              target="_blank"
            >
              https://gitlab.dummy.endpoint.com
            </a>
          </td>
          <td
            className="pf-c-table__action"
            data-key={3}
            data-label={null}
            onMouseEnter={[Function]}
            style={
              Object {
                "paddingRight": 0,
              }
            }
          >
            <div
              className="pf-c-dropdown pf-m-align-right"
              data-ouia-component-id="OUIA-Generated-Dropdown-6"
              data-ouia-component-type="PF4/Dropdown"
              data-ouia-safe={true}
            >
              <button
                aria-expanded={false}
                aria-haspopup={true}
                aria-label="Actions"
                className="pf-c-dropdown__toggle pf-m-plain"
                disabled={true}
                id="pf-dropdown-toggle-id-5"
                onClick={[Function]}
                onKeyDown={[Function]}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style={
                    Object {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 192 512"
                  width="1em"
                >
                  <path
                    d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                  />
                </svg>
              </button>
            </div>
          </td>
        </tr>
        <tr
          className=""
          data-ouia-component-id="OUIA-Generated-TableRow-9"
          data-ouia-component-type="PF4/TableRow"
          data-ouia-safe={true}
          hidden={false}
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <td
            className="pf-c-table__check"
            data-key={0}
            data-label={null}
            onMouseEnter={[Function]}
          >
            <label>
              <input
                aria-label="Select row 2"
                checked={false}
                disabled={true}
                name="checkrow2"
                onChange={[Function]}
                type="checkbox"
              />
            </label>
          </td>
          <td
            className=""
            data-key={1}
            data-label="Name"
            onMouseEnter={[Function]}
          >
            Bitbucket
            <span
              data-testid="bitbucket-unsupported-automatic-revoke-tooltip"
              data-tip="Provided API does not support the automatic token revoke. You can revoke it manually."
            >
              <svg
                aria-hidden={true}
                aria-labelledby={null}
                fill="var(--pf-global--warning-color--100)"
                height="1em"
                role="img"
                style={
                  Object {
                    "margin": "2px 5px",
                    "verticalAlign": "text-top",
                  }
                }
                viewBox="0 0 1088 1024"
                width="1em"
              >
                <path
                  d="M1057.10141,663.5 L845.101405,215.4 C787.101405,71.8 665.401405,0 542.901405,0 C420.201405,0 296.701405,71.9 235.001405,215.6 L31.7014051,648.5 C10.4014051,700 -0.0985948775,752.3 0.000697596367,800.8 C0.301405123,924.8 70.2014051,1024 209.101405,1024 L868.401405,1024 C1005.80141,1024 1087.70141,918.6 1088.00215,795.5 C1088.10141,752.4 1078.20141,707.2 1057.10141,663.5 Z M959.401405,800.3 C958.701405,822.9 952.901405,843.5 942.601405,859.7 C926.801405,884.6 902.601405,896.7 868.301405,896.7 L209.101405,896.7 C191.201405,896.7 176.601405,893.8 165.401405,888.2 C157.301405,884 150.801405,878.4 145.401405,870.3 C135.101405,855 129.101405,832 128.401405,805.6 C127.601405,772.8 134.901405,736.5 149.401405,700.5 L353.001405,266.7 C363.201405,242.9 376.101405,221.5 391.101405,203.2 C404.801405,186.6 420.301405,172.4 437.401405,161.1 C469.201405,139.9 505.701405,128.8 542.901405,128.8 C579.701405,128.8 615.401405,139.8 646.001405,160.5 C662.401405,171.6 677.101405,185.4 690.101405,201.6 C704.501405,219.6 716.401405,240.6 725.901405,264 L940.901405,718.9 L941.101405,719.3 L941.301405,719.7 C953.901405,746 960.201405,773.9 959.401405,800.3 Z M586.601405,832 L501.301405,832 C489.501405,831.8 480.201405,821.5 480.001405,808.7 L480.001405,727.3 C480.201405,714.5 489.601405,704.3 501.301405,704 L586.601405,704 C598.401405,704.2 607.701405,714.5 607.901405,727.3 L607.901405,808.7 L608.001405,808.7 C607.701405,821.5 598.301405,831.8 586.601405,832 M639.901405,290.7 L613.201405,610.4 C611.801405,626.9 598.001405,640 581.301405,640 L506.601405,640 C490.001405,640 476.101405,627.2 474.701405,610.7 L448.101405,291 C446.501405,272.3 461.301405,256.3 480.001405,256.3 L608.001405,256 C626.701405,256 641.401405,272 639.901405,290.7"
                />
              </svg>
            </span>
            <div>
              Dummy Tooltip
            </div>
          </td>
          <td
            className=""
            data-key={2}
            data-label="Server"
            onMouseEnter={[Function]}
          >
            <a
              href="https://bitbucket.dummy.endpoint.org"
              rel="noreferrer"
              target="_blank"
            >
              https://bitbucket.dummy.endpoint.org
            </a>
          </td>
          <td
            className="pf-c-table__action"
            data-key={3}
            data-label={null}
            onMouseEnter={[Function]}
            style={
              Object {
                "paddingRight": 0,
              }
            }
          >
            <div
              className="pf-c-dropdown pf-m-align-right"
              data-ouia-component-id="OUIA-Generated-Dropdown-7"
              data-ouia-component-type="PF4/Dropdown"
              data-ouia-safe={true}
            >
              <button
                aria-expanded={false}
                aria-haspopup={true}
                aria-label="Actions"
                className="pf-c-dropdown__toggle pf-m-plain"
                disabled={true}
                id="pf-dropdown-toggle-id-6"
                onClick={[Function]}
                onKeyDown={[Function]}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style={
                    Object {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 192 512"
                  width="1em"
                >
                  <path
                    d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                  />
                </svg>
              </button>
            </div>
          </td>
        </tr>
        <tr
          className=""
          data-ouia-component-id="OUIA-Generated-TableRow-10"
          data-ouia-component-type="PF4/TableRow"
          data-ouia-safe={true}
          hidden={false}
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <td
            className="pf-c-table__check"
            data-key={0}
            data-label={null}
            onMouseEnter={[Function]}
          >
            <label>
              <input
                aria-label="Select row 3"
                checked={false}
                disabled={true}
                name="checkrow3"
                onChange={[Function]}
                type="checkbox"
              />
            </label>
          </td>
          <td
            className=""
            data-key={1}
            data-label="Name"
            onMouseEnter={[Function]}
          >
            Microsoft Azure DevOps
            <span
              data-testid="azure-devops-unsupported-automatic-revoke-tooltip"
              data-tip="Provided API does not support the automatic token revoke. You can revoke it manually."
            >
              <svg
                aria-hidden={true}
                aria-labelledby={null}
                fill="var(--pf-global--warning-color--100)"
                height="1em"
                role="img"
                style={
                  Object {
                    "margin": "2px 5px",
                    "verticalAlign": "text-top",
                  }
                }
                viewBox="0 0 1088 1024"
                width="1em"
              >
                <path
                  d="M1057.10141,663.5 L845.101405,215.4 C787.101405,71.8 665.401405,0 542.901405,0 C420.201405,0 296.701405,71.9 235.001405,215.6 L31.7014051,648.5 C10.4014051,700 -0.0985948775,752.3 0.000697596367,800.8 C0.301405123,924.8 70.2014051,1024 209.101405,1024 L868.401405,1024 C1005.80141,1024 1087.70141,918.6 1088.00215,795.5 C1088.10141,752.4 1078.20141,707.2 1057.10141,663.5 Z M959.401405,800.3 C958.701405,822.9 952.901405,843.5 942.601405,859.7 C926.801405,884.6 902.601405,896.7 868.301405,896.7 L209.101405,896.7 C191.201405,896.7 176.601405,893.8 165.401405,888.2 C157.301405,884 150.801405,878.4 145.401405,870.3 C135.101405,855 129.101405,832 128.401405,805.6 C127.601405,772.8 134.901405,736.5 149.401405,700.5 L353.001405,266.7 C363.201405,242.9 376.101405,221.5 391.101405,203.2 C404.801405,186.6 420.301405,172.4 437.401405,161.1 C469.201405,139.9 505.701405,128.8 542.901405,128.8 C579.701405,128.8 615.401405,139.8 646.001405,160.5 C662.401405,171.6 677.101405,185.4 690.101405,201.6 C704.501405,219.6 716.401405,240.6 725.901405,264 L940.901405,718.9 L941.101405,719.3 L941.301405,719.7 C953.901405,746 960.201405,773.9 959.401405,800.3 Z M586.601405,832 L501.301405,832 C489.501405,831.8 480.201405,821.5 480.001405,808.7 L480.001405,727.3 C480.201405,714.5 489.601405,704.3 501.301405,704 L586.601405,704 C598.401405,704.2 607.701405,714.5 607.901405,727.3 L607.901405,808.7 L608.001405,808.7 C607.701405,821.5 598.301405,831.8 586.601405,832 M639.901405,290.7 L613.201405,610.4 C611.801405,626.9 598.001405,640 581.301405,640 L506.601405,640 C490.001405,640 476.101405,627.2 474.701405,610.7 L448.101405,291 C446.501405,272.3 461.301405,256.3 480.001405,256.3 L608.001405,256 C626.701405,256 641.401405,272 639.901405,290.7"
                />
              </svg>
            </span>
            <div>
              Dummy Tooltip
            </div>
          </td>
          <td
            className=""
            data-key={2}
            data-label="Server"
            onMouseEnter={[Function]}
          >
            <a
              href="https://azure.dummy.endpoint.com/"
              rel="noreferrer"
              target="_blank"
            >
              https://azure.dummy.endpoint.com/
            </a>
          </td>
          <td
            className="pf-c-table__action"
            data-key={3}
            data-label={null}
            onMouseEnter={[Function]}
            style={
              Object {
                "paddingRight": 0,
              }
            }
          >
            <div
              className="pf-c-dropdown pf-m-align-right"
              data-ouia-component-id="OUIA-Generated-Dropdown-8"
              data-ouia-component-type="PF4/Dropdown"
              data-ouia-safe={true}
            >
              <button
                aria-expanded={false}
                aria-haspopup={true}
                aria-label="Actions"
                className="pf-c-dropdown__toggle pf-m-plain"
                disabled={true}
                id="pf-dropdown-toggle-id-7"
                onClick={[Function]}
                onKeyDown={[Function]}
                type="button"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style={
                    Object {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 192 512"
                  width="1em"
                >
                  <path
                    d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
                  />
                </svg>
              </button>
            </div>
          </td>
        </tr>
      </tbody>
    </table>
  </section>,
]
`;

exports[`GitServices should correctly render the component without git services 1`] = `
Array [
  <span
    className="progressLine"
  >
    
  </span>,
  <section
    className="pf-c-page__main-section"
  >
    <div
      className="pf-c-empty-state pf-m-sm pf-m-full-height"
    >
      <div
        className="pf-c-empty-state__content"
      >
        <svg
          aria-hidden="true"
          aria-labelledby={null}
          className="pf-c-empty-state__icon"
          fill="currentColor"
          height="1em"
          role="img"
          style={
            Object {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 1152 1024"
          width="1em"
        >
          <path
            d="M504.601506,583.6 C499.901506,576.8 493.701506,572 485.901506,569.1 L270.001506,450.5 C265.501506,448.8 261.001506,448 256.401506,448 L255.701506,448 L255.001506,448 C250.601506,448 246.001506,448.8 241.601506,450.5 L25.7015061,569.2 C18.0015061,572.1 11.7015061,576.9 7.00150606,583.7 C2.30150606,590.4 0.00074731781,598 0.00074731781,606.2 L0.00074731781,862.6 C-0.098493943,879.8 9.70150606,886 20.4015061,892 L236.401506,1019.1 C242.101506,1022.4 248.401506,1024 255.101506,1024 L255.801506,1024 L256.501506,1024 C263.201506,1024 269.501506,1022.4 275.201506,1019.1 L491.201506,892 C501.401506,886.5 511.701506,879.8 511.701506,865.4 L511.701506,606.1 C511.701506,597.9 509.301506,590.4 504.601506,583.6 Z M474.801506,832 L255.801506,960 L35.6015061,832 L35.6015061,768 L255.701506,896 L474.801506,768 L474.801506,832 Z M474.801506,704 L255.801506,832 L35.6015061,704 L35.6015061,640 L255.701506,768 L474.801506,640 L474.801506,704 Z M1144.60151,583.6 C1139.90151,576.8 1133.70151,572 1125.90151,569.1 L910.001506,450.5 C905.501506,448.8 901.001506,448 896.401506,448 L895.701506,448 L895.001506,448 C890.601506,448 886.001506,448.8 881.601506,450.5 L665.701506,569.2 C658.001506,572.1 651.701506,576.9 647.001506,583.7 C642.301506,590.4 640.001506,598 640.001506,606.2 L640.001506,862.6 C640.001506,879.8 649.801506,886 660.501506,892 L876.501506,1019.1 C882.201506,1022.4 888.501506,1024 895.201506,1024 L895.901506,1024 L896.601506,1024 C903.301506,1024 909.601506,1022.4 915.301506,1019.1 L1131.30151,892 C1141.50151,886.5 1151.80151,879.8 1151.80151,865.4 L1151.80151,606.1 C1151.70151,597.9 1149.30151,590.4 1144.60151,583.6 Z M1114.80151,832 L895.701506,960 L675.601506,832 L675.601506,768 L895.701506,896 L1114.80151,768 L1114.80151,832 Z M1114.80151,704 L895.701506,832 L675.601506,704 L675.601506,640 L895.701506,768 L1114.80151,640 L1114.80151,704 Z M824.601506,95.6 C819.901506,88.8 813.701506,84 805.901506,81.1 L590.001506,2.5 C585.501506,0.8 581.001506,0 576.401506,0 L575.701506,0 L575.001506,0 C570.601506,0 566.001506,0.8 561.601506,2.5 L345.701506,81.2 C338.001506,84.1 331.701506,88.9 327.001506,95.7 C322.401506,102.4 320.001506,110 320.001506,118.2 L320.001506,359.6 C320.001506,376.8 329.801506,383 340.501506,389 L556.501506,507.1 C562.201506,510.4 568.501506,512 575.201506,512 L575.901506,512 L576.601506,512 C583.301506,512 589.601506,510.4 595.301506,507.1 L811.301506,389 C821.501506,383.5 831.801506,376.8 831.801506,362.4 L831.801506,118.1 C831.701506,109.9 829.301506,102.4 824.601506,95.6 Z M794.801506,332.3 L575.801506,448 L355.601506,332.3 L355.601506,272 L575.701506,384 L794.801506,272 L794.801506,332.3 Z M794.801506,208 L575.801506,320 L355.601506,208 L355.601506,144 L575.701506,256 L794.801506,144 L794.801506,208 Z"
          />
        </svg>
        <h4
          className="pf-c-title pf-m-lg"
        >
          No Git Services
        </h4>
      </div>
    </div>
  </section>,
]
`;
